cli/js/tests/io_test.ts

```diff
@@ -55,3 +55,19 @@ unitTest(async function copyWithCustomBufferSize() {
   assertEquals(write.length, xBytes.length);
   assertEquals(readSpy.calls, DEFAULT_BUF_SIZE / bufSize + 1);
 });
+
+unitTest({ perms: { write: true } }, async function copyBufferToFile() {
+  const filePath = "test-file.txt";
+  // bigger than max File possible buffer 16kb
+  const bufSize = 32 * 1024;
+  const xBytes = repeat("b", bufSize);
+  const reader = new Deno.Buffer(xBytes.buffer as ArrayBuffer);
+  const write = await Deno.open(filePath, { write: true, create: true });
+
+  const n = await Deno.copy(reader, write, { bufSize });
+
+  assertEquals(n, xBytes.length);
+
+  write.close();
+  await Deno.remove(filePath);
+});
```

Comment:
This is in fact failing on latest master - it seems there a bug in `Deno.copy()` - less bytes may be written than read on each iteration.

Reply:
I'm doing a rewrite on this PR, I'll loop inside `copy` to make sure `.write` actually writes the whole buffer.